### PR TITLE
Gets rid of for-loop deprecation compiler warnings in stdlib/public/core

### PR DESCRIPTION
**Pull request by kovpas**
Not too many warnings, but still rather annoying.

### CODE REVIEW

File: lib/Sema/TypeCheckType.cpp [hunk 1]
```diff
@@ -367,8 +367,10 @@ Type TypeChecker::applyGenericArguments(Type type,
   if (!unbound) {
     // FIXME: Highlight generic arguments and introduce a Fix-It to remove
     // them.
-    diagnose(loc, diag::not_a_generic_type, type);
-
+    if (!type->is<ErrorType>()) {
```
Comment (DougGregor) [inline lib/Sema/TypeCheckType.cpp]: This should be part of a separate pull request.

Comment (kovpas) [inline lib/Sema/TypeCheckType.cpp]: These are not my changes. Probably rebase issues. Fixing.

**Comment by DougGregor** [comment]
Can you separate out the generic-specialization-of-error-type changes?

**Comment by kovpas** [comment]
@DougGregor that was a rebase issue, I've already fixed this.

**Comment by gribozavr** [comment]
We had applied such a change before, and it caused performance issues.  We can safely take all changes outside of `stdlib/public/core`, though.

@nadavrot, how would you like to proceed with this?

@kovpas Feel free to split the patch into the `stdlib/public/core` and "the rest".  We will immediately merge the latter.

**Comment by kovpas** [comment]
@gribozavr done: https://github.com/apple/swift/pull/653
This pull request now contains changes in `stdlib/public/core`. The other one is "the rest".

**Comment by gribozavr** [comment]
I believe this is already implemented in master.
